From 7fbd7c25c4010e8e5fbbc0d8741cff9c69bcea05 Mon Sep 17 00:00:00 2001
From: Vadim Pasternak <vadimp@nvidia.com>
Date: Thu, 7 Oct 2021 19:30:05 +0000
Subject: [PATCH backport 5.10 082/182] mlxsw: core: Extend external cooling
 device whitelist for thermal zone binding

Allow thermal zone binding to an external cooling device of type
"emc2305".
Motivation to support new system SN2021 equipped with ASIC Spectrum-1
supporting 48x Gbps RJ45 + 4x100G QSFP28 ports.
System airflow control is provided by EMC2305 RPM-based PWM Fan Speed
Controller as colling device.

Signed-off-by: Vadim Pasternak <vadimp@nvidia.com>
---
 drivers/net/ethernet/mellanox/mlxsw/core_thermal.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/drivers/net/ethernet/mellanox/mlxsw/core_thermal.c b/drivers/net/ethernet/mellanox/mlxsw/core_thermal.c
index e1a760519097..91abc7a3f7ea 100644
--- a/drivers/net/ethernet/mellanox/mlxsw/core_thermal.c
+++ b/drivers/net/ethernet/mellanox/mlxsw/core_thermal.c
@@ -30,6 +30,7 @@
 /* External cooling devices, allowed for binding to mlxsw thermal zones. */
 static char * const mlxsw_thermal_external_allowed_cdev[] = {
 	"mlxreg_fan",
+	"emc2305",
 };
 
 enum mlxsw_thermal_trips {
-- 
2.20.1

